aptos-move/framework/src/natives/cryptography/algebra/constants.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright © Aptos Foundation
+
+use crate::{
+    abort_unless_arithmetics_enabled_for_structure, abort_unless_feature_flag_enabled,
+    natives::{
+        cryptography::algebra::{
+            feature_flag_from_structure, gas::GasParameters, AlgebraContext, Structure,
+            BLS12381_GT_GENERATOR, BLS12381_R_LENDIAN, MOVE_ABORT_CODE_NOT_IMPLEMENTED,
+        },
+        helpers::{SafeNativeContext, SafeNativeError, SafeNativeResult},
+    },
+    store_element, structure_from_ty_arg,
+};
+use ark_ec::Group;
+use move_core_types::gas_algebra::NumArgs;
+use move_vm_types::{loaded_data::runtime_types::Type, values::Value};
+use num_traits::{One, Zero};
+use once_cell::sync::Lazy;
+use smallvec::{smallvec, SmallVec};
+use std::{collections::VecDeque, rc::Rc};
+
+macro_rules! ark_constant_op_internal {
+    ($context:expr, $ark_typ:ty, $ark_func:ident, $gas:expr) => {{
+        $context.charge($gas)?;
+        let new_element = <$ark_typ>::$ark_func();
+        let new_handle = store_element!($context, new_element);
+        Ok(smallvec![Value::u64(new_handle as u64)])
+    }};
+}
+
+pub fn zero_internal(
```

Comment:
Decided to use additive group notations across the module, and further let groups and fields "share" some ops (e.g., `add`, `one`, `zero`) to make the module more concise. @alinush is a fan of this.

That said, I agree it's confusing that `zero()` returns 1 for a multiplication group. And I don't see any issues switching back to standalone group APIs like `group_identity()`/`group_generator()`.

I'm thinking to leave it as-is for now and see community feedback that's coming soon.

Reply:
I don't think it's confusing because a multiplicative group can be viewed as an additive group. That's the premise here!

Plus, the Move dev is agnostic of the representation of group elements. So even though for prime order subgroups of Z_p^* the identity is (can be viewed as) 1, this module will never expose the number 1 as the return value of zero(); it is handle-based, which means devs don't get to see what's behind the returned Element<T>.